File: WorldFeed/Server/src/WorldFeed.Common/Helpers/IHttpClientWebHelper.cs
namespace WorldFeed.Common.Helpers
{
    using System.Net.Http;
    using System.Threading.Tasks;

    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Web;

    public interface IHttpClientWebHelper
    {
        Task<HttpResponseMessage> GetHttpResponseAsync(ITwitterQuery twitterQuery, ITwitterClientHandler handler = null);

        HttpClient GetHttpClient(ITwitterQuery twitterQuery, ITwitterClientHandler handler = null);
    }
}

File: WorldFeed/Client/src/app/core/Core/Helpers/IHttpClientWebHelper.ts
import {ITwitterQuery} from "../../Public/Models/Interfaces/ITwitterQuery";
import {ITwitterClientHandler} from "../Web/ITwitterClientHandler";
import {HttpClient} from "@aspnet/signalr";

export interface IHttpClientWebHelper {
  getHttpResponseAsync(twitterQuery: ITwitterQuery, handler: ITwitterClientHandler /*= null*/): Promise<HttpResponseMessage>;

  getHttpClient(twitterQuery: ITwitterQuery, handler: ITwitterClientHandler /*= null*/): HttpClient;
}
